packages/open-payments/src/client/incoming-payment.test.ts

```diff
@@ -322,6 +340,156 @@ describe('incoming-payment', (): void => {
     })
   })
 
+  describe('listIncomingPayment', (): void => {
+    const paymentPointer = `${baseUrl}/.well-known/pay`
+
+    describe('forward pagination', (): void => {
+      test.each`
+        first        | cursor
+        ${undefined} | ${undefined}
+        ${1}         | ${undefined}
+        ${5}         | ${uuid()}
+      `(
+        'returns incoming payments list',
+        async ({ first, cursor }): Promise<void> => {
+          const incomingPaymentPaginationResult =
+            mockIncomingPaymentPaginationResult({
+              result: Array(first).fill(mockIncomingPayment())
+            })
+
+          const scope = nock(paymentPointer)
+            .get('/incoming-payments')
+            .query({
+              ...(first ? { first } : {}),
+              ...(cursor ? { cursor } : {})
+            })
+            .reply(200, incomingPaymentPaginationResult)
+
+          const result = await listIncomingPayment(
+            {
+              axiosInstance,
+              logger
+            },
+            {
+              paymentPointer,
+              accessToken: 'accessToken'
+            },
+            openApiValidators.successfulValidator,
+            {
+              first,
+              cursor
+            }
+          )
+
+          expect(result).toStrictEqual(incomingPaymentPaginationResult)
+          scope.done()
+        }
+      )
+    })
+
+    describe('backward pagination', (): void => {
+      test.each`
+        last         | cursor
+        ${undefined} | ${uuid()}
+        ${5}         | ${uuid()}
```

Comment:
This test is setup correctly, but in general makes me confused about the backward pagination parameters. 

If we only provide a `cursor`, like (backwards-pagination allows) then by default we will end up forwards paginating, based on this code in the `backend`:

https://github.com/interledger/rafiki/blob/b19919a68d1ff782193d171d83053579b5abdff2/packages/backend/src/shared/pagination.ts#L9-L20

would this mean that we should make `last` required as well in Open Payments API? Only `cursor` is required:
 
![image](https://user-images.githubusercontent.com/15069181/207863011-a9d9446b-5b93-443d-9464-dcf2b80c20d4.png)

Reply:
:grimacing: 
related:
- https://github.com/interledger/open-payments/issues/233
- https://github.com/interledger/open-payments/issues/136

Reply:
I'm in favour of _not_ supporting both parameters (for now) to make the Open Payments pagination type similar to what you describe in [comment](https://github.com/interledger/open-payments/pull/129#discussion_r887840847):

`after + count` || `before + count`